### ISSUE TITLE
Error building on android

### ISSUE THREAD

**Issue by samin-safaei** [BlueChilli/flutter_uploader#15]
When I added the plugin with the required configuration in `build.gradle` I get the following error while building:
```
Execution failed for task ':app:preDebugBuild'.
> Android dependency 'androidx.arch.core:core-runtime' has different version for the compile (2.0.0) and runtime (2.0.1) classpath. You should manually set the same version via DependencyResolution
```
Here is the result of `flutter doctor -v` command:
```
[✓] Flutter (Channel stable, v1.7.8+hotfix.4, on Linux, locale en_US.UTF-8)
    • Flutter version 1.7.8+hotfix.4 at /home/conscript/Projects/flutter
    • Framework revision 20e59316b8 (5 weeks ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /home/conscript/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /home/conscript/Android/Sdk
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b16-5323222)
    • All Android licenses accepted.
[✓] Android Studio (version 3.4)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 37.0.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b16-5323222)
[✓] IntelliJ IDEA Ultimate Edition (version 2018.3)
    • IntelliJ at /opt/idea-IU-183.4588.48
    • Flutter plugin version 32.0.2
    • Dart plugin version 183.4588.61
[✓] Connected device (1 available)
    • SM G950F • ce03171328d3ac340c • android-arm64 • Android 9 (API 28)
• No issues found!
```
I have the following dependencies in my project:
```
  http: ^0.12.0
  flutter_svg: ^0.13.1
  flutter_slidable: ^0.4.9
  location: ^2.3.5
  flutter_auth_buttons: ^0.3.1
  flutter_range_slider: ^1.0.1
  shared_preferences: ^0.5.2
  uni_links: ^0.2.0
  flutter_spinkit: ^2.0.1
  image_picker: ^0.6.0+17
  cached_network_image: ^1.0.0
  intl: ^0.15.8
  app_settings: ^1.0.6
  permission_handler: ^3.0.1
  google_sign_in: ^4.0.4
  firebase_auth: ^0.11.1+12
  cloud_firestore: ^0.12.9
  firebase_admob: ^0.9.0+3
  firebase_messaging: ^5.1.2
  firebase_analytics: ^4.0.2
  firebase_crashlytics: ^0.0.4+11
  flutter_image_compress: ^0.6.3
  image_cropper: ^1.0.1
  flutter_pagewise: ^1.2.3
  cupertino_icons: ^0.1.0
  keyboard_visibility: ^0.5.6
  local_auth: ^0.5.2+3
  flutter_sticky_header:
    git: https://github.com/smm13344331/flutter_sticky_header.git
  crypto: ^2.0.6
  logging: ^0.11.3+2
  overlay_support: ^1.0.1
  vibration: ^1.2.1
  flutter_ringtone_player: ^1.0.3
```

**Comment by onikiri2007** [comment]
I believe this would solve your problem 
https://github.com/flutter/flutter/issues/27254.

**Comment by m-jabrane** [comment]
Hello,

For anyone who's having the same issue building a release app in Android. I added the required configuration, described in `Readme`, into my `build.gradle` (in `android` folder), but it doesn't work.

Using:
```
cloud_firestore: ^0.14.3+1
flutter_uploader: ^2.0.0-beta.3
```
It shows the error below:
```
Building plugin cloud_firestore...
The plugin cloud_firestore could not be built due to the issue above.
```

So for my case, I just changed the value `android.enableR8`, in my `gradle.properties` file, from `false` to `true` without any additional configurations (previous configurations described in `Readme` removed) to my app, and it worked.

My `gradle.properties` file:
```
org.gradle.jvmargs=-Xmx1536M
android.useAndroidX=true
android.enableJetifier=true
android.enableR8=true
```

Hope it helps someone!
